### DISCUSSION TITLE
Sharing caches or sharing client?

### DISCUSSION THREAD

**Discussion by numbers1311407** [TanStack/query#4147] [Q&A]
Is it appropriate to share MutationCache and QueryCache between clients? Or if not, it is appropriate to share a client between apps?

The project I'm on is a client side micro frontend which mounts react packages separately, so they don't share a common QueryClientProvider.

I started down the path of sharing a single queryClient cross app, as that seemed to be the intention of query client context sharing, but realized the QC provider setup doesn't support this out of the box because the single client would be mounted/unmounted in the lifecycle of each provider.

This made me wonder if this sharing the caches was more appropriate, it seems like it may be but I thought I'd look for expert advice before continuing on either path.

**Comment by TkDodo** [comment]
the client is just a vessel for the caches, which holds default configs. you can share either one.
